crypto/engine/eng_table.c

```diff
@@ -217,7 +217,10 @@ ENGINE *ossl_engine_table_select(ENGINE_TABLE **table, int nid,
     }
     ERR_set_mark();
     if (!CRYPTO_THREAD_write_lock(global_engine_lock))
+    {
+        lock_err = 1;
         goto end;
+    }
```

Comment:
As above, if you move ERR_set_mark below this check, you can just return NULL on a failed write lock, and avoid having to track this variable separately.